pkg/controller/factory/builder.go

```diff
@@ -73,10 +73,13 @@ func BuildRSM(cluster *appsv1alpha1.Cluster, synthesizedComp *component.Synthesi
 	mergeAnnotations := intctrlutil.MergeMetadataMaps(constant.GetKBGenerationAnnotation(synthesizedComp.ClusterGeneration),
 		getMonitorAnnotations(synthesizedComp), synthesizedComp.Annotations)
 
+	replicasStr := strconv.Itoa(int(synthesizedComp.Replicas))
 	podBuilder := builder.NewPodBuilder("", "").
+		AddLabelsInMap(synthesizedComp.Labels).
```

Comment:
cannot directly set pod labels here,  Any upgrades or label updates will result in unexpected pod restarts.

Reply:
rsm2 will support pod in-place update for fields `annotations`, `labels`, `image` etc. which can avoid pod recreation.

Reply:
> rsm2 will support pod in-place update for fields `annotations`, `labels`, `image` etc. which can avoid pod recreation.

How about the pods that don't use rsm2? Can they also be compatible?

Reply:
`custom labels` and `custom annotations` are static and immutable from the ComponentDefinition API description.
So it's safe to put them into the `PodTemplate`.

Reply:
> ComponentDefinition API

The code here utilizes synthesizedComp.Labels, which comprise annotations and labels from several parts. One part originates from the ComponentDefinition API, while another part is inherited from the cluster. The labels and annotations inherited from the cluster are dynamic and can change.